common/chat.ts

```diff
@@ -2,17 +2,17 @@ import { Jsonify } from './json'
 import { SbUser, SbUserId } from './users/sb-user'
 
 export enum ChatServiceErrorCode {
-  UserOffline = 'UserOffline',
-  UserNotFound = 'UserNotFound',
-  InvalidJoinAction = 'InvalidJoinAction',
+  AlreadyJoined = 'AlreadyJoined',
   LeaveShieldBattery = 'LeaveShieldBattery',
-  InvalidLeaveAction = 'InvalidLeaveAction',
-  InvalidModerationAction = 'InvalidModerationAction',
-  ModeratorAccess = 'ModeratorAccess',
+  ModerateChannelOwner = 'ModerateChannelOwner',
```

Comment:
I think these are better when they're phrased as the error, not as what you were trying to do. "CannotModerateYourself" is better than "ModerateYourself", etc.